### ISSUE TITLE
Recent version of conda-build (3.18.11) fails to build with nonsensical conflict messages

### ISSUE THREAD

**Issue by klauer** [pcdshub/pcdsdevices#399]
Failing version:
```
          conda version : 4.8.2
    conda-build version : 3.18.11
```

Working version:
```
          conda version : 4.8.2
    conda-build version : 3.12.0
```

Conflicts such as:
```
Output in format: Requested package -> Available versions

Package pyyaml conflicts for:

bluesky[version='>=1.2.0'] -> databroker -> pyyaml

pyyaml

Package setuptools conflicts for:

python[version='>=3.6'] -> pip -> setuptools

ophyd[version='1.4.0rc4|>=1.4.0'] -> networkx -> setuptools

matplotlib -> setuptools

pyepics[version='>=3.4.1'] -> pip -> setuptools

bluesky[version='>=1.2.0'] -> ipython -> setuptools[version='>=18.5']

pytmc[version='>=2.4.0'] -> jinja2 -> setuptools
```

See https://github.com/pcdshub/pcdsdevices/pull/396 and https://travis-ci.org/github/pcdshub/pcdsdevices/jobs/661344045  for more information

**Comment by ZLLentz** [comment]
The conflict output also included dependencies for python 3.8 (noticed by @ZryletTC), despite us building for 3.6 and 3.7 only, which tipped us off that it must be a conda issue.

**Comment by klauer** [comment]
We aren't pinning conda-build and we aren't having this issue currently.
